src/cdsetool/download.py

```diff
@@ -158,3 +164,31 @@ def _get_credentials(options: Dict) -> Credentials:
 
 def _get_temp_dir(options: Dict) -> Union[str, None]:
     return options.get("tmpdir") or None
+
+
+def checksum_for_file(
+    file_path: str, file_name: str, proxy: Union[Dict[str, str], None] = None
+) -> bool:
+    """Evaluate the checksum of a file. If the file was downloaded correctly"""
+    odata_for_checksum = get_odata_by_name(file_name, proxy)
+    array_values = odata_for_checksum.get("value", [])
+    if len(array_values) == 0:
+        # no data available for checksum
```

Comment:
Do you know if there are a lot of products missing checksums now so this return is something expected even when things "work"?

Reply:
From my side I don't know if Copernicus resolved all the problems regarding the checksums. I remember there where some issues on older products missing the checksum or having an incorrect one  but not sure if things are now resolved. I guess it's better to still check if that "value" field is present than not to check it.